src/git-hosting-api/git-hosting-api.ts

```diff
@@ -0,0 +1,33 @@
+export interface GitHostingUserResponse {
+  login: string;
+  name: string;
+  html_url: string;
+}
+
+export interface GitHostingIssueResponse {
+  number: number;
+  title: string;
+  pull_request?: {
+    html_url: string;
+  };
+  labels: Array<{
+    name: string;
+  }>;
+  user: {
+    login: string;
+    html_url: string;
+  };
+}
+
+export interface Options {
+  repo: string;
+  rootPath: string;
+  cacheDir?: string;
+  gitHostingServerURL: string;
+}
+
+export interface GitHostingAPI {
+  getBaseIssueUrl(repo: string): string;
+  getIssueData(repo: string, issue: string): Promise<GitHostingIssueResponse>;
+  getUserData(login: string): Promise<GitHostingUserResponse>;
+}
```

Comment:
This is an abstracted interface. If this interface is satisfied, there is no need to pay attention to specific implementations in other parts of the application.